packages/strapi-plugin-content-manager/controllers/EditingLock.js

```diff
@@ -0,0 +1,179 @@
+'use strict';
+
+const _ = require('lodash');
+const {
+  getService,
+} = require('../utils');
+const { validateLockInput, validateExtendLockInput } = require('./validation');
+
+const sanitizeLock = (lock, { withUid = false } = {}) => {
+  if (_.isNil(lock)) {
+    return null;
+  }
+
+  const omittedAttributes = ['key', 'id'];
+  if (!withUid) {
+    omittedAttributes.push('uid');
+  }
+
+  return _.omit(lock, omittedAttributes);
+};
+
+const getLock = async ctx => {
+  const {
+    state: { userAbility },
+    params: { id, model },
+  } = ctx;
+
+  const { kind } = strapi.getModel(model);
+
+  const entityManager = getService('entity-manager');
+  const permissionChecker = getService('permission-checker').create({ userAbility, model });
+  const editingLockService = getService('editing-lock');
+
+  if (permissionChecker.cannot.update()) {
+    return ctx.forbidden();
+  }
+
+  const entity = await entityManager.findOne(id, model);
+
+  if (!entity && kind !== 'singleType') {
```

Comment:
In case of a single type you can return not found in the case there is a create permission too (can check the find ST controller)

Reply:
My logic for single-types is to be able to take the lock even if en entity doesn't exist, so i would not throw here. What do you think ?